client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -367,7 +368,8 @@ private void throwIfClosed() throws TxnFailedException {
         
         Map<Segment, SegmentTransaction<Type>> transactions = new HashMap<>();
         for (Segment s : segments.getSegments()) {
-            SegmentOutputStream out = outputStreamFactory.createOutputStreamForTransaction(s, txId, config, segments.getDelegationToken());
+            SegmentOutputStream out = outputStreamFactory.createOutputStreamForTransaction(s, txId, config,
+                    DelegationTokenProviderFactory.create(segments.getDelegationToken(), controller, s));
```

Comment:
Same here.

Reply:
Reusing the delegation token provider as suggested.